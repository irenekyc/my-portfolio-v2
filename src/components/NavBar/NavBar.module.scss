.ikNavBar {
  position: fixed;
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8%;
  background: #3a4750;
  z-index: 999;

  &__container {
    width: 100%;
    max-width: 1100px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    h2 {
      height: 60px;
      line-height: 60px;
    }
  }

  &__hamburger {
    &__div {
      height: 60px;
      width: 60px;
      display: flex;
      -webkit-box-pack: center;
      -ms-flex-pack: center;
      justify-content: center;
      -webkit-box-align: center;
      -ms-flex-align: center;
      align-items: center;
      input {
        visibility: hidden;
      }
    }

    &__button {
      width: 50%;
      height: 50%;
      display: -webkit-box;
      display: -ms-flexbox;
      display: flex;
      -webkit-box-orient: vertical;
      -webkit-box-direction: normal;
      -ms-flex-direction: column;
      flex-direction: column;
      -ms-flex-pack: distribute;
      justify-content: space-around;
      -webkit-box-align: center;
      -ms-flex-align: center;
      align-items: center;
      -webkit-transition: all 1s;
      transition: all 1s;
      cursor: pointer;
      &__line {
        width: 80%;
        height: 3px;
        background: #00adb5;
        -webkit-transition: all 1s;
        transition: all 1s;
      }
    }
  }
}

.ikNavBar__open {
  -webkit-transform: rotate(135deg);
  transform: rotate(135deg);

  .ikNavBar__hamburger__button__line {
    background: #eee;
    &:nth-child(3) {
      display: none;
    }
    &:nth-child(2) {
      transform: rotate(270deg);
    }
    &:nth-child(4) {
      transform: translate3d(0, -7px, 0);
    }
  }
}
