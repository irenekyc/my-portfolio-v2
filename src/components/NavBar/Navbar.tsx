import React, { useState } from "react";
import styles from "./NavBar.module.scss";
import ModalMenu from "../ModalMenu";

const NavBar = () => {
  const [isOpen, setIsOpen] = useState<boolean>(false);

  return (
    <>
      <nav className={styles.ikNavBar}>
        <h1> Irene K. </h1>
        <div className={styles.ikNavBar__hamburger__div}>
          <label
            htmlFor="toggle-menu"
            className={`${isOpen ? styles.ikNavBar__open : ""} ${
              styles.ikNavBar__hamburger__button
            }`}
          >
            <input
              type="checkbox"
              value="open"
              id="toggle-menu"
              onClick={() => setIsOpen(!isOpen)}
            />
            <div className={styles.ikNavBar__hamburger__button__line}></div>
            <div className={styles.ikNavBar__hamburger__button__line}></div>
            <div className={styles.ikNavBar__hamburger__button__line}></div>
          </label>
        </div>
      </nav>
      {<ModalMenu showMenu={isOpen} />}
    </>
  );
};

export default NavBar;
