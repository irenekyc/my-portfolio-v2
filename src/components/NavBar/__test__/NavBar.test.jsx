import React from "react";
import NavBar from "../index";
import { render, screen, fireEvent } from "@testing-library/react";

it("Render NavBar Section", () => {
  render(<NavBar showMenu={false} />);

  const navBar = screen.getByTestId("nav-bar");
  expect(navBar).toBeInTheDocument();
  const hamburgerBtn = screen.getByTestId("hamburger-btn");
  expect(hamburgerBtn).not.toHaveClass("ikNavBar__open");
  expect(navBar).toMatchSnapshot();
});

it("Render NavBar Section - click on hamburger button", () => {
  render(<NavBar showMenu={false} />);

  const navBar = screen.getByTestId("nav-bar");
  expect(navBar).toBeInTheDocument();
  const hamburgerBtn = screen.getByTestId("hamburger-btn");
  fireEvent.click(hamburgerBtn);
  expect(hamburgerBtn).toHaveClass("ikNavBar__open");
});
