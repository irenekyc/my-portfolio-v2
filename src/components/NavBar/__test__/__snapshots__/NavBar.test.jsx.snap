// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavBar Section 1`] = `
<nav
  class="ikNavBar"
  data-testid="nav-bar"
>
  <div
    class="ikNavBar__container"
  >
    <h2>
       Irene K. 
    </h2>
    <div
      class="ikNavBar__hamburger__div"
    >
      <label
        class=" ikNavBar__hamburger__button"
        data-testid="hamburger-btn"
        for="toggle-menu"
      >
        <input
          id="toggle-menu"
          type="checkbox"
          value="open"
        />
        <div
          class="ikNavBar__hamburger__button__line"
        />
        <div
          class="ikNavBar__hamburger__button__line"
        />
        <div
          class="ikNavBar__hamburger__button__line"
        />
      </label>
    </div>
  </div>
</nav>
`;
